components/cam/src/physics/cam/shoc_intr.F90

```diff
@@ -0,0 +1,1258 @@
+   real(r8) :: thv2(pcols,pver)  
+ 
+   real(r8) :: minqn, rrho(pcols,pver), rrho_i(pcols,pverp)    ! minimum total cloud liquid + ice threshold    [kg/kg]
+   real(r8) :: cldthresh, frac_limit
+   real(r8) :: ic_limit, dum1
+ 
+   real(r8) :: wpthlp_sfc(pcols), wprtp_sfc(pcols), upwp_sfc(pcols), vpwp_sfc(pcols)
+   real(r8) :: wtracer_sfc(pcols,edsclr_dim)
+  
+   ! Variables below are needed to compute energy integrals for conservation
+   real(r8) :: ke_a(pcols), ke_b(pcols), te_a(pcols), te_b(pcols)
+   real(r8) :: wv_a(pcols), wv_b(pcols), wl_b(pcols), wl_a(pcols)
+   real(r8) :: se_dis(pcols), se_a(pcols), se_b(pcols), shoc_s(pcols,pver)
+   real(r8) :: shoc_t(pcols,pver)
+   
+   ! --------------- !
+   ! Pointers        !
+   ! --------------- !
+  
+   real(r8), pointer, dimension(:,:) :: tke  ! turbulent kinetic energy 
+   real(r8), pointer, dimension(:,:) :: wthv ! buoyancy flux
+   real(r8), pointer, dimension(:)   :: tauresx
+   real(r8), pointer, dimension(:)   :: tauresy
+   real(r8), pointer, dimension(:,:) :: tkh 
+   real(r8), pointer, dimension(:,:) :: tk
+   real(r8), pointer, dimension(:,:) :: cld      ! cloud fraction                               [fraction]
+   real(r8), pointer, dimension(:,:) :: concld   ! convective cloud fraction                    [fraction]
+   real(r8), pointer, dimension(:,:) :: ast      ! stratiform cloud fraction                    [fraction]
+   real(r8), pointer, dimension(:,:) :: alst     ! liquid stratiform cloud fraction             [fraction]
+   real(r8), pointer, dimension(:,:) :: aist     ! ice stratiform cloud fraction                [fraction] 
+   real(r8), pointer, dimension(:,:) :: cmeliq 
+           
+   real(r8), pointer, dimension(:,:) :: qlst     ! Physical in-stratus LWC                      [kg/kg]
+   real(r8), pointer, dimension(:,:) :: qist     ! Physical in-stratus IWC                      [kg/kg]
+   real(r8), pointer, dimension(:,:) :: deepcu   ! deep convection cloud fraction               [fraction]
+   real(r8), pointer, dimension(:,:) :: shalcu   ! shallow convection cloud fraction            [fraction]    
+   real(r8), pointer, dimension(:,:) :: khzt     ! eddy diffusivity on thermo levels            [m^2/s]
+   real(r8), pointer, dimension(:,:) :: khzm     ! eddy diffusivity on momentum levels          [m^2/s]
+   real(r8), pointer, dimension(:) :: pblh     ! planetary boundary layer height                [m]
+   real(r8), pointer, dimension(:,:) :: dp_icwmr ! deep convection in cloud mixing ratio        [kg/kg] 
+   real(r8), pointer, dimension(:,:) :: cmfmc_sh ! Shallow convective mass flux--m subc (pcols,pverp) [kg/m2/s/]     
+
+   real(r8), pointer, dimension(:,:) :: prer_evap 
+   real(r8), pointer, dimension(:,:) :: accre_enhan
+   real(r8), pointer, dimension(:,:) :: relvar
+   
+   logical :: lqice(pcnst)   
+
+#endif
+   det_s(:)   = 0.0_r8
+   det_ice(:) = 0.0_r8
```

Comment:
It looks like here you want to make sure det_s and det_ice are defined for the output designation even if SHOC_SGS is not defined.  But what about ptend_all and alst_o which are also outputs.  Should these be initialized as well?

Reply:
was just following the CLUBB interface.  Those declarations were unnecessary because if SHOC is not activated, it will never make it into the shoc_tend_e3sm routine.  Will delete this portion of code.